compat_test/smoke_compat_test.go

```diff
@@ -0,0 +1,189 @@
+		rpc.WithWebsocketMessageSizeLimit(1024 * 1024 * 256),
+	}
+)
+
+func init() {
+	flag.Uint64Var(&blockInterval, "block-interval", 1_000_000, "number of blocks to skip between test iterations")
+	flag.BoolVar(&enableRandomBlockTest, "random-block", false, "enable random block height within the range during test iterations")
+}
+
+func TestSmokeRPCCompatibilities(t *testing.T) {
+	// TestSmokeRPCCompatibilities checks the compatibility of Celo L1 and Celo L2 RPC responses.
+	// The test retrieves the following types of data at the beginning, the last L1 block, and at every one million blocks to verify compatibility.
+	// 1. Block
+	// 2. Transactions
+	// 3. Transaction Receipts
+	// 4. Block Receipts
+	flag.Parse()
+
+	if celoRpcURL == "" {
+		t.Fatal("celo rpc url not set example usage:\n go test -v ./compat_test -tags compat_test -run TestSmokeRPCCompatibilities -celo-url ws://localhost:8546 -op-geth-url ws://localhost:9994")
+	}
+	if opGethRpcURL == "" {
+		t.Fatal("op-geth rpc url not set example usage:\n go test -v ./compat_test -tags compat_test -run TestSmokeRPCCompatibilities -celo-url ws://localhost:8546 -op-geth-url ws://localhost:9994")
+	}
+
+	// Setup RPC clients for Celo L1 and Celo L2 server
+	celoClient, err := rpc.DialOptions(context.Background(), celoRpcURL, clientOpts...)
+	require.NoError(t, err)
+	celoEthClient := ethclient.NewClient(celoClient)
+
+	opClient, err := rpc.DialOptions(context.Background(), opGethRpcURL, clientOpts...)
+	require.NoError(t, err)
+	opEthClient := ethclient.NewClient(opClient)
+
+	initCtx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	t.Cleanup(cancel)
+
+	// Fetch Chain IDs and make sure they're same
+	celoChainId, err := celoEthClient.ChainID(initCtx)
+	require.NoError(t, err)
+	opChainId, err := opEthClient.ChainID(initCtx)
+	require.NoError(t, err)
+	require.Equal(t, celoChainId.Uint64(), opChainId.Uint64(), "chain ids of referenced chains differ")
+
+	// Make sure the Chain ID is supported
+	chainId := celoChainId.Uint64()
+	_, ok := gingerbreadBlocks[chainId]
+	require.True(t, ok, "chain id %d not found in supported chainIDs %v", chainId, gingerbreadBlocks)
+
+	// Fetch the last block height in Celo L1
+	lastCeloL1BlockHeight, err := celoEthClient.BlockNumber(initCtx)
+	require.NoError(t, err)
+
+	t.Logf("Last Celo L1 Block Height: %d", lastCeloL1BlockHeight)
+
+	resultChan := make(chan *blockResults, 100)
+	globalCtx, cancel := context.WithCancel(context.Background())
+	t.Cleanup(cancel)
+
+	// Fetching RPC data job
+	fetchingEg, jobCtx := errgroup.WithContext(globalCtx)
+	fetchingEg.SetLimit(6)
+
+	fetchingEg.Go(func() error {
+		clients := &clients{
+			celoEthclient: celoEthClient,
+			opEthclient:   opEthClient,
+			celoClient:    celoClient,
+			opClient:      opClient,
+		}
+
+		// Fetch some random blocks between 0 to lastCeloL1BlockHeight
+		for currentBlockNumber := uint64(0); currentBlockNumber < lastCeloL1BlockHeight; currentBlockNumber += blockInterval {
```

Comment:
I think it would be good to check both the 0 and number 1 block, since they are created/stored in quite different ways.
```suggestion
		// Fetch some random blocks between 0 to lastCeloL1BlockHeight
		asyncFetchRpcData(t, jobCtx, fetchingEg, 0, clients, resultChan)
		for currentBlockNumber := uint64(1); currentBlockNumber < lastCeloL1BlockHeight; currentBlockNumber += blockInterval {
```